### PR TITLE
display shapes in specific position

### PR DESCRIPTION
**Pull request by fearblackcat**
please help merge

### CODE REVIEW

File: debug/js/main.js [hunk 1]
```diff
@@ -14,8 +14,21 @@ $$.Board = class Board {
 
   paint(elements) {
     elements.map(item => {
-      this._ctx.fillRect(item.x, item.y, item.w, item.h)
-      this._ctx.fillText(item.id, item.x, item.y)
+      if(item.type == 'Image' && item.id == 'image1') {
```
Comment (yinrong) [inline debug/js/main.js]: this is debug tool, you should work on lib/DynamicCreative, not here!

File: lib/DynamicCreative.js [hunk 2]
```diff
@@ -1,42 +1,26 @@
...
+    this._canvas = canvas;
+    this._elements = [];
   }
 
   add(elements) {
     // TODO implement this
+    this._elements = elements;
   }
 
   toParams(canvas) {
```
Comment (yinrong) [inline lib/DynamicCreative.js]: you're not following README, this is supposed to be renamed

**Comment by fearblackcat** [comment]
Please help review again. Thanks

File: debug/js/main.js [hunk 3]
```diff
@@ -5,18 +5,18 @@ if (!window.$$) {
 
 $$.Board = class Board {
   constructor(opts) {
-    this._dom = document.createElement("canvas")
-    this._dom.width = opts.w
-    this._dom.height = opts.h
-    document.getElementById("board").appendChild(this._dom)
-    this._ctx = this._dom.getContext("2d")
+    this._dom = document.createElement("canvas");
```
Comment (yinrong) [inline debug/js/main.js]: please keep original style

File: lib/DynamicCreative.js [hunk 4]
```diff
@@ -1,42 +1,54 @@
...
-        y: 284.6666666666667,
-        w: 192,
-        h: 45 },
-    ]
+    let scalecanvas = canvas;
+    let newElements = [];
+    let originalElements = this._elements;
+    let containerWidth = canvas.w;
+    let boxSpacing = 15;
+    let layoutGeometry = require('justified-layout')(this._aspects, {
```
Comment (yinrong) [inline lib/DynamicCreative.js]: not following the coding style attached in the email.

Comment (yinrong) [reply lib/DynamicCreative.js]: it is good manner to search for third-party packages. but you still ignore the real requirement and background, etc. the results in the debug page are far from good.

**Comment by yinrong** [comment]
image

your results are not similar to this illustration at all.

**Comment by fearblackcat** [comment]
Maybe this time I would have satisfied the requirements need.
Please help review again. Thanks!

**Comment by fearblackcat** [comment]
Please help review again. Thanks!

**Pull request closed without merging**